dvc/command/version.py

```diff
@@ -1,29 +1,62 @@
 from __future__ import unicode_literals
 
+import os
+import psutil
 import platform
 import argparse
 import logging
 
+from dvc.repo import Repo
 from dvc.command.base import CmdBaseNoRepo, append_doc_link
 from dvc.version import __version__
+from dvc.exceptions import NotDvcRepoError
 
 
 logger = logging.getLogger(__name__)
 
 
 class CmdVersion(CmdBaseNoRepo):
     def run(self):
-        info = (
-            "DVC version: {dvc_version}\n"
-            "Python version: {python_version}\n"
-            "Platform: {platform_type}"
-        ).format(
-            dvc_version=__version__,
-            python_version=platform.python_version(),
-            platform_type=platform.platform(),
-        )
-        logger.info(info)
-        return 0
+        try:
+            root_directory = Repo.find_root()
+
+            info = (
+                "DVC version: {dvc_version}\n"
+                "Python version: {python_version}\n"
+                "Platform: {platform_type}\n"
+                "Filesystem Type: {filesystem_type}\n"
+                "Symlink: {symlink}"
+            ).format(
+                dvc_version=__version__,
+                python_version=platform.python_version(),
+                platform_type=platform.platform(),
+                filesystem_type=self.get_fs_type(
+                    os.path.abspath(root_directory)
+                ),
+                symlink=os.path.islink(os.path.abspath(root_directory)),
```

Comment:
Sorry, I might be missing something, but why do we need to know if repo itself is a symlink? The original issue is about testing whether or not reflink, hardlink and symlink are supported from cache directory to the workspace. You can test that by createing a temporary file in `dvc.cache.local.cache_dir` and trying to `System.{ref, sym, hard}link`(from dvc/system.py) it to `dvc.root_dir`.

Reply:
@efiop Yes, I might have committed a mistake here but that's because I don't understand why do we need to know filesystem and link type info?

Reply:
@algomaster99 Ah, got it :slightly_smiling_face: Sorry, totally forgot to properly explain the logic behind it. So dvc needs to somehow link cached files to your workspace. It is able to utilize 4 types of links: reflink, hardlink, symlink, copy. From those four only `copy` is generally guaranteed to work in any circumstances, all others have their limitations:

1) reflink - supported by a limited number of filesystems(e.g. APFS, XFS, Btrfs, etc) and doesn't work across filesystems like symlink does(this scenario is useful when you are storing your dvc cache on an an external hdd while you are working on your dvc project on SSD. See https://dvc.org/doc/use-cases/multiple-data-scientists-on-a-single-machine for a bit more context); 
2) hardlink - doesn't work across filesystems;
3) symlink - might not work across filesystems, if filesystems are not mounted with proper flags;

So to know which links are supported, we need to go ahead and try creating them. :slightly_smiling_face:

Reply:
@algomaster99 check also this link https://github.com/iterative/dvc/issues/1599, read the issue and an expandable section in it.